### PR TITLE
docs(BPM API): Split the big page into smaller ones

### PR DESCRIPTION
**Pull request by vbotnari12**
The page is too heavy for crawler algolia, hence we are going to split it into small pages.
RUNTIME-849 : [DOC] Split BPM api page in multiple sub pages to make it indexable by algolia

### CODE REVIEW

**Comment by educhastenier** [comment]
That would be clearer if all API pages had `api` in their name.

**Comment by educhastenier** [comment]
The new pages are not in the index, is it wanted? (ie. they do not appear on the left side menu)

**Comment by vbotnari12** [comment]
> The new pages are not in the index, is it wanted? (ie. they do not appear on the left side menu)

how do we do it?

**Comment by educhastenier** [comment]
> > The new pages are not in the index, is it wanted? (ie. they do not appear on the left side menu)
> 
> how do we do it?

@vbotnari12 you modify taxonomy.adoc to add the new pages under bpm-api.adoc.

File: modules/ROOT/pages/bpm-api.adoc [hunk 1]
```diff
@@ -3,5150 +3,9 @@
...
- ** java.util.Date (Supports *ISO-8601* format, eg: `2013-01-02T02:42:12.17+02:00`)
- ** java.time.LocalDate (Supports *ISO-8601* format, eg: `2018-02-04`)
- ** java.time.LocalDateTime (Supports *ISO-8601* format, eg: `2018-02-04T10:30:15`)
- ** java.time.OffsetDateTime (Supports *ISO-8601* format, eg: `2018-02-04T10:30:15+01:00`) +
-When not set, the type is guessed using the value. Be careful as it can lead to type inconsistency in the target process (eg: a java.lang.Long is expected and the guessed type is a java.lang.Integer)
-
-* *Success Response*
- ** *Code*: 204
+* xref:activities-tasks-flow-nodes-api.adoc[Activities, Tasks and Flow nodes]
+* xref:data-case-document-api.adoc[Data and CaseDocument]
```
Comment (educhastenier) [inline modules/ROOT/pages/bpm-api.adoc]: The commonly use term is simply Document and not CaseDocument, even if the technical name of the api is caseDocument:
```suggestion
* xref:data-case-document-api.adoc[Data and Document]
```

File: modules/ROOT/pages/data-case-document-api.adoc [hunk 2]
```diff
@@ -0,0 +1,629 @@
+= Data and Case Document
```
Comment (educhastenier) [inline modules/ROOT/pages/data-case-document-api.adoc]: Same remark:
```suggestion
= Data and Documents 
```

**Review by educhastenier — CHANGES_REQUESTED** [review]
remove Case from 'Case Document'

File: modules/ROOT/pages/data-case-document-api.adoc (hunk 2 above)
Comment (vbotnari12) [reply modules/ROOT/pages/data-case-document-api.adoc]: Done. Thanks

File: modules/ROOT/taxonomy.adoc [hunk 3]
```diff
@@ -228,8 +228,12 @@
   *** xref:rest-api-authentication.adoc[REST Authentication]
   *** xref:application-api.adoc[Application API]
   *** xref:access-control-api.adoc[Access control API]
-  *** xref:bdm-api.adoc[Bdm API]
-  *** xref:bpm-api.adoc[Bpm API]
+  *** xref:bdm-api.adoc[BDM API]
+  *** xref:bpm-api.adoc[BPM API]
+   **** xref:activities-tasks-flow-nodes-api.adoc[Activities, Tasks and Flow nodes]
+   **** xref:data-document-api.adoc[Data and Document]
```
Comment (educhastenier) [inline modules/ROOT/taxonomy.adoc]:
```suggestion
   **** xref:data-document-api.adoc[Data and Documents]
```

File: modules/ROOT/pages/bpm-api.adoc [hunk 4]
```diff
@@ -3,5150 +3,9 @@
 
 Easily retrieve human or service tasks, call activities, and subprocesses with BPM APIs. Perform maintenance tasks.
 
-== Activities and Tasks
+Documentation split :
```
Comment (educhastenier) [inline modules/ROOT/pages/bpm-api.adoc]: One last remark / question to you @vbotnari12 : Do we want to keep this in the page?

File: modules/ROOT/pages/bpm-api.adoc [hunk 5]
```diff
@@ -3,5150 +3,9 @@
...
- ** java.util.Date (Supports *ISO-8601* format, eg: `2013-01-02T02:42:12.17+02:00`)
- ** java.time.LocalDate (Supports *ISO-8601* format, eg: `2018-02-04`)
- ** java.time.LocalDateTime (Supports *ISO-8601* format, eg: `2018-02-04T10:30:15`)
- ** java.time.OffsetDateTime (Supports *ISO-8601* format, eg: `2018-02-04T10:30:15+01:00`) +
-When not set, the type is guessed using the value. Be careful as it can lead to type inconsistency in the target process (eg: a java.lang.Long is expected and the guessed type is a java.lang.Integer)
-
-* *Success Response*
- ** *Code*: 204
+* xref:activities-tasks-flow-nodes-api.adoc[Activities, Tasks and Flow nodes]
+* xref:data-document-api.adoc[Data and Document]
```
Comment (educhastenier) [inline modules/ROOT/pages/bpm-api.adoc]:
```suggestion
* xref:data-document-api.adoc[Data and Documents]
```

**Review by educhastenier — APPROVED** [review]

**Pull request merged**